### PR TITLE
Add instructor's default availability

### PR DESCRIPTION
**Pull request by bibixx**
## What part of the app is affected?
* [x] Frontend
* [x] Backend
* [ ] Infrastructure

### CODE REVIEW

File: backend/src/defaultAvailability/entities/defaultAvailability.entity.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+/**
+ * @see https://github.com/typeorm/typeorm/issues/2390
+ */
+const transformFromTimeToDatabaseType = (value: Time | FindOperator<any>) => {
+  if (value instanceof FindOperator) {
+    const oldOperator = value as unknown as { _value: Time };
+    const newOperator = value as unknown as { _value: string };
+
+    // eslint-disable-next-line no-underscore-dangle
```
Comment (maonat) [inline backend/src/defaultAvailability/entities/defaultAvailability.entity.ts:20]: @bibixx  Shall we fix the eslint rule if we want to keep the code "clean" from comments?

Comment (bibixx) [reply backend/src/defaultAvailability/entities/defaultAvailability.entity.ts:20]: Well, the property with an underscore is an internal property of typeorm which we can't really change. That's why I have had to disable that rule for this line

File: shared/src/global.d.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// eslint-disable-next-line @typescript-eslint/ban-types
```
Comment (maonat) [inline shared/src/global.d.ts:1]: Same here, change eslint rules? @bibixx

File: shared/src/types/Time.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+  static fromString(value: string) {
+    const [hours, minutes, seconds] = value.split(':');
+    const hoursAsNumber = Number.parseInt(hours!, 10);
+    const minutesAsNumber = Number.parseInt(minutes!, 10);
+    const secondsAsNumber = Number.parseInt(seconds!, 10);
+    return new Time(hoursAsNumber, minutesAsNumber, secondsAsNumber);
+  }
+
+  // eslint-disable-next-line @typescript-eslint/ban-types
```
Comment (maonat) [inline shared/src/types/Time.ts:16]: @bibixx  eslint types

File: shared/src/global.d.ts (hunk 2 above)
Comment (bibixx) [reply shared/src/global.d.ts:1]: That is kind of expected and an exception to the rule, where we _generally_ shouldn't use `Date` in the shared folder, but it sometimes is necessary.

**Review by maonat — APPROVED** [review]

**Pull request merged**